libraries/AP_RCTelemetry/AP_CRSF_Telem.h

```diff
@@ -0,0 +1,227 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+
+#ifndef HAL_CRSF_TELEM_ENABLED
+#define HAL_CRSF_TELEM_ENABLED !HAL_MINIMIZE_FEATURES
+#endif
+
+#if HAL_CRSF_TELEM_ENABLED
+
+#include <AP_Notify/AP_Notify.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_HAL/utility/RingBuffer.h>
+#include <AP_RCProtocol/AP_RCProtocol_CRSF.h>
+#include "AP_RCTelemetry.h"
+#include <AP_HAL/utility/sparse-endian.h>
+
+class AP_CRSF_Telem : public AP_RCTelemetry {
+public:
+    AP_CRSF_Telem();
+    ~AP_CRSF_Telem() override;
+
+    /* Do not allow copies */
+    AP_CRSF_Telem(const AP_CRSF_Telem &other) = delete;
+    AP_CRSF_Telem &operator=(const AP_CRSF_Telem&) = delete;
+
+    // init - perform required initialisation
+    virtual bool init() override;
+
+    static AP_CRSF_Telem *get_singleton(void);
+
+    // Broadcast frame definitions courtesy of TBS
+    struct GPSFrame {   // curious fact, calling this GPS makes sizeof(GPS) return 1!
+        int32_t latitude; // ( degree / 10`000`000 )
+        int32_t longitude; // (degree / 10`000`000 )
+        uint16_t groundspeed; // ( km/h / 100 )
+        uint16_t gps_heading; // ( degree / 100 )
+        uint16_t altitude; // ( meter - 1000m offset )
+        uint8_t satellites; // in use ( counter )
+    } PACKED;
```

Comment:
I don't think we should mark a structure packed unless it is really needed (has misaligned elements). It just makes for larger and slower code

Reply:
Ok, I am going to call you on this :) With packing on a Pixracer:
```
Target          Text     Data  BSS     Total
----------------------------------------------
bin/arducopter  1338072  1476  195352  1534900
```
without packing of all these structures (so won't work, but just for comparison):
```
Target          Text     Data  BSS     Total
----------------------------------------------
bin/arducopter  1338060  1476  195356  1534892
```
So text goes down and bss goes up giving a total of 8 bytes difference. The code being slow-er is pretty much irrelevant here - I get the convenience of casting a byte stream to a structure and reading values, I suspect the overhead of looping through the bytes is going to be pretty much the same. And the cognitive dissonance of trying to figure out which are safe and which are not seems not worth the effort.

But happy to be corrected as always!